src/tests/class_tests/openms/source/PeakIntegrator_test.cpp

```diff
@@ -135,6 +141,69 @@ constexpr const char* BASELINE_TYPE_BASETOBASE = "base_to_base";
+  1808499.25, 3368120.75, 3803323.25, 4059358, 4092095, 4058075.25, 4160395, 4395341.5, 4573185, 4565417.5, 4371225, 4065336.75, 3716669.5, 3338531, 2982165.25, 2675112, 2464135, 2307864, 2178321, 2053773.88, 1941723.12, 1851044.12, 1768557.75, 1694074, 1643185, 1621222.5, 1615733.12, 1586897.38, 1506051.75, 1374752.25, 1228128.12, 1104698.5, 1004596.56, 922247.438, 846399.25, 775016.375, 717635, 678566.562, 656717.375, 645997, 640481.062, 637876.188, 632300.688, 619110.375, 599831.625, 578701.5, 559403.312, 543297.625, 528049.25, 509227.594, 485023.156, 462585.688, 448690.75, 438401.25, 425110.969, 409034.406, 394403.406, 381114.688, 369241.094, 356554.969, 345889.281, 336646.844, 325948.938, 316341.688, 310746.156, 309641.875
+};
+
+const vector<double> cutoff_pos_min = {
+  15.34253311, 15.35624981, 15.36995029, 15.38366699, 15.39736652, 15.41156673, 15.42574978, 15.44018364, 15.45436668, 15.46856689, 15.48274994, 15.49695015
+};
+
+const vector<double> cutoff_pos_sec = {
+  920.5519866, 921.3749885999999, 922.1970174, 923.0200194, 923.8419912, 924.6940038, 925.5449868000001, 926.4110184000001, 927.2620008, 928.1140134, 928.9649964, 929.817009
+};
+
+const vector<double> cutoff_int = {
+  3.48297429, 15.54384613, 50.31319046, 151.8971405, 411.25631714, 946.44311523, 1642.56152344, 2118.89526367, 2055.13647461, 1665.13232422, 1275.53015137, 1009.70056152
+};
+
+MSChromatogram saturated_chrom_min, saturated_cutoff_chrom_min, cutoff_chrom_min;
+MSChromatogram saturated_chrom_sec, saturated_cutoff_chrom_sec, cutoff_chrom_sec;
+MSSpectrum saturated_spec_min, saturated_cutoff_spec_min, cutoff_spec_min;
+MSSpectrum saturated_spec_sec, saturated_cutoff_spec_sec, cutoff_spec_sec;
+for (Size i = 0; i < saturated_pos_min.size(); ++i)
+{
+  saturated_chrom_min.push_back(ChromatogramPeak(saturated_pos_min[i], saturated_int[i]));
+  saturated_chrom_sec.push_back(ChromatogramPeak(saturated_pos_sec[i], saturated_int[i]));
+  saturated_spec_min.push_back(Peak1D(saturated_pos_min[i], saturated_int[i]));
+  saturated_spec_sec.push_back(Peak1D(saturated_pos_sec[i], saturated_int[i]));
+}
+for (Size i = 0; i < saturated_cutoff_pos_min.size(); ++i)
+{
+  saturated_cutoff_chrom_min.push_back(ChromatogramPeak(saturated_cutoff_pos_min[i], saturated_cutoff_int[i]));
+  saturated_cutoff_chrom_sec.push_back(ChromatogramPeak(saturated_cutoff_pos_sec[i], saturated_cutoff_int[i]));
+  saturated_cutoff_spec_min.push_back(Peak1D(saturated_cutoff_pos_min[i], saturated_cutoff_int[i]));
+  saturated_cutoff_spec_sec.push_back(Peak1D(saturated_cutoff_pos_sec[i], saturated_cutoff_int[i]));
+}
+for (Size i = 0; i < cutoff_pos_min.size(); ++i)
+{
+  cutoff_chrom_min.push_back(ChromatogramPeak(cutoff_pos_min[i], cutoff_int[i]));
+  cutoff_chrom_sec.push_back(ChromatogramPeak(cutoff_pos_sec[i], cutoff_int[i]));
+  cutoff_spec_min.push_back(Peak1D(cutoff_pos_min[i], cutoff_int[i]));
+  cutoff_spec_sec.push_back(Peak1D(cutoff_pos_sec[i], cutoff_int[i]));
+}
+
+
```

Comment:
Just to double check, you also include the "ideal" peak in your tests?

Reply:
Yes. In those cases, theidentifiers are: `chromatogram`, `spectrum`, `position` and `intensity`, depending on which data structure I'm testing (a `MSChromatogram`, `MSSpectrum`, and `double`s)